File: sdks/csharp/src/NovaBilling/Analytics/Requests/GetRevenueAnalyticsRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record GetRevenueAnalyticsRequest
{
    [JsonIgnore]
    public string? DateFrom { get; set; }

    [JsonIgnore]
    public string? DateTo { get; set; }

    [JsonIgnore]
    public string? Currency { get; set; }

    [JsonIgnore]
    public GetRevenueAnalyticsRequestGroupBy? GroupBy { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/Analytics/Requests/GetRevenueAnalyticsRequest.php
<?php

namespace NovaBilling\Analytics\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Analytics\Types\GetRevenueAnalyticsRequestGroupBy;

class GetRevenueAnalyticsRequest extends JsonSerializableType
{
    /**
     * @var ?string $dateFrom
     */
    public ?string $dateFrom;

    /**
     * @var ?string $dateTo
     */
    public ?string $dateTo;

    /**
     * @var ?string $currency
     */
    public ?string $currency;

    /**
     * @var ?value-of<GetRevenueAnalyticsRequestGroupBy> $groupBy
     */
    public ?string $groupBy;

    /**
     * @param array{
     *   dateFrom?: ?string,
     *   dateTo?: ?string,
     *   currency?: ?string,
     *   groupBy?: ?value-of<GetRevenueAnalyticsRequestGroupBy>,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->dateFrom = $values['dateFrom'] ?? null;
        $this->dateTo = $values['dateTo'] ?? null;
        $this->currency = $values['currency'] ?? null;
        $this->groupBy = $values['groupBy'] ?? null;
    }
}
